File: Sheet 3 (Loops 1)/R.py
t = int(input())
for i in range(t):
    a , b, c , n = input().split()
    a = int(a); b = int(b) ; c = int(c) ; n =int(n)
    maxi =max(max(a,b) , c)
    sm = 0
    sm += maxi - a
    sm += maxi - b
    sm += maxi - c
    n -= sm
    if n % 3 == 0 and n >= 0 :
        print("YES")
    else:
        print("NO")
    

File: Sheet 3 (Loops 1)/R.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){

    int t;
    cin >> t;
    while (t--){
        long long a , b , c , n , maxi = -1 , co = 0 ;
        cin >> a >> b >> c >> n;
        maxi = max(a , max(b , c));
        co += maxi - a;
        co += maxi - b;
        co += maxi - c;
        n -= co;
        if (n >= 0 && n % 3 == 0)cout <<"YES\n";
        else cout <<"NO\n";

    }
}
